### ISSUE TITLE
Firefox Focus crashes on `i` info icon opening

### ISSUE THREAD

**Issue by Nickoriginal** [mozilla-mobile/focus-android#5022]
## Steps to reproduce
1. Open any page in Firefox Focus.
2. Open three-dot menu.
3. Tap on the `i` info icon opposite the "Trackers blocked" menu.
### Expected behavior
Firefox Focus should open "Focus FAQ" page and not crash.
### Actual behavior
After several taps Focus crashes and sends crash report to Mozilla.

https://user-images.githubusercontent.com/85299944/127208077-1d94fe98-113c-4f9c-bb98-b3c14a32fe6f.mp4

### Device information 

* Android device: Samsung Galaxy A20s, Android 11
* Focus version: 8.18.0-rc.1

**Comment by pocmo** [comment]
The menu doesn't close and then we try to invoke something on the context of the fragment that is gone:

```
                         E  java.lang.IllegalStateException: Fragment BrowserFragment{762474e} (f730d042-07f0-47f0-b420-173e00e563d3) not attached to a context.
                         E      at androidx.fragment.app.Fragment.requireContext(Fragment.java:900)
                         E      at org.mozilla.focus.fragment.BrowserFragment.onClick(BrowserFragment.kt:636)
                         E      at org.mozilla.focus.menu.browser.BlockingItemViewHolder._init_$lambda-0(BlockingItemViewHolder.kt:43)
                         E      at org.mozilla.focus.menu.browser.BlockingItemViewHolder.$r8$lambda$ZtbP2EOXnGRc36lYE27CVJ8d-Kg(Unknown Source:0)
                         E      at org.mozilla.focus.menu.browser.BlockingItemViewHolder$$ExternalSyntheticLambda0.onClick(Unknown Source:2)
                         E      at android.view.View.performClick(View.java:7441)
                         E      at android.view.View.performClickInternal(View.java:7418)
                         E      at android.view.View.access$3700(View.java:835)
                         E      at android.view.View$PerformClick.run(View.java:28675)
                         E      at android.os.Handler.handleCallback(Handler.java:938)
                         E      at android.os.Handler.dispatchMessage(Handler.java:99)
                         E      at android.os.Looper.loopOnce(Looper.java:201)
                         E      at android.os.Looper.loop(Looper.java:288)
                         E      at android.app.ActivityThread.main(ActivityThread.java:7829)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
```

**Comment by lobontiumira** [comment]
Verified as fixed on the 91.1.0 Focus build using Google Pixel (Android 10), and OnePlus 5T (Android 10).
